### PR TITLE
Set up a single SensorTask

### PR DESCRIPTION
**Pull request by newpomax**

### CODE REVIEW

File: src/quail/sensors/SensorTask.cpp [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+void SensorTask::activity(){
+    xEventGroupWaitBits(evgroup, ADC_STARTED, pdTRUE, pdFALSE, NEVER); // wait for ADC to begin (don't clear bits)
+    for(uint8_t i = 0; i < Sensor::numSensors(); i++){
+        sys.sensors[i]->configure();
+    }
+    sys.tasks.adctask.sensorsConfigured();
+    while(true){
+        uint32_t flags = xEventGroupWaitBits(evgroup, ADC_RESET|DATA_READY, pdTRUE, pdFALSE, NEVER); // await a reset or a data ready flag from ADC
+        if(flags & ADC_RESET)
+            xEventGroupWaitBits(evgroup, ADC_STARTED, pdTRUE, pdFALSE, NEVER);
```
Comment (Timvrakas) [inline src/quail/sensors/SensorTask.cpp:16]: Does this properly handle reconfiguration? I think we need to break, and then I don't know if the sensor objects handle reconfiguring correctly.

**Review by Timvrakas — COMMENTED** [review]
- Needs to properly handle ADC failure, reset, and reconfiguration
- Should test on hardware (with bag of ice lol) before merging. @mikh3x4 @polygnomial ?

**Comment by Timvrakas** [comment]
Also, I don't know if it makes sense to have the SensorTask be a Task, as opposed to a Sensors class that holds Sensor objects. I think the time requires to switch tasks is significantly longer than the time needed to compute the raw ADC data into float sensor values, so the ADC task could just call the function to perform the operation on the sensor data and have it all live within a single task.  1 tasks > 2 tasks > N+2 tasks

**Comment by newpomax** [comment]
> Also, I don't know if it makes sense to have the SensorTask be a Task, as opposed to a Sensors class that holds Sensor objects. I think the time requires to switch tasks is significantly longer than the time needed to compute the raw ADC data into float sensor values, so the ADC task could just call the function to perform the operation on the sensor data and have it all live within a single task. 1 tasks > 2 tasks > N+2 tasks

I'm fine with that - would the ADCTask also be triggering all the setup stuff directly? That also seems more clear and removes all of that event-bit flagging

**Comment by theDarkKnight5050** [comment]
Is context switching slower even with us collecting all the sensor data before processing? I'd figured since we're doing one bulk computation every N interrupts it's still worth doing.

**Pull request merged**